src/project_model.rs

```diff
@@ -65,15 +65,15 @@ impl ProjectModel {
         let resolve = resolve_with_prev(&mut registry, &ws, prev.as_ref())?;
         let cargo_packages = {
             let ids: Vec<PackageId> = resolve.iter().cloned().collect();
-            registry.get(&ids)
+            registry.get(&ids)?
         };
         let mut pkg_id_to_pkg = HashMap::new();
         let mut manifest_to_id = HashMap::new();
         let mut packages = Vec::new();
         for (idx, pkg_id) in resolve.iter().enumerate() {
             let pkg = Package(idx);
             pkg_id_to_pkg.insert(pkg_id.clone(), pkg);
-            let cargo_pkg = cargo_packages.get(pkg_id)?;
+            let cargo_pkg = cargo_packages.get_one(pkg_id)?;
```

Comment:
Note that the breakge here stems from https://github.com/rust-lang/cargo/pull/6005 where Cargo can now download crates in parallel. The `get_one` is the (now renamed) old synchronous interface, and RLS likely wants to try using the new [`get_many`](https://github.com/rust-lang/cargo/blob/04cec5fc10a7bbe1dccaad5729eff1823ef2608c/src/cargo/core/package.rs#L349-L362) interface to leverage parallel downloads. For now though I've left it as-is to hopefully get the regression fixed quickly, and this could perhaps be a future follow-up!

Reply:
Maybe leave a FIXME or open an issue so this isn't forgotten?